xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-default/src/main/java/org/xwiki/model/validation/internal/EntityNameValidationConfigurationListener.java

```diff
@@ -0,0 +1,119 @@
+
+/**
+ * Listen for create/update/delete events on the entity name validation classes and objects. When their values change,
+ * clears the entity name validators caches (by calling {@link EntityNameValidation#cleanConfigurationCache(String)} on
+ * the registered {@link EntityNameValidation} components). The cache is managed per wiki, and only the cache of the
+ * current wiki is cleared.
+ *
+ * @version $Id$
+ * @since 13.4
+ * @since 12.10.8
+ */
+@Component
+@Named(EntityNameValidationConfigurationListener.NAME)
+@Singleton
+public class EntityNameValidationConfigurationListener implements EventListener
+{
+    /**
+     * The name of the event listener.
+     */
+    static final String NAME = "EntityNameValidationConfigurationListener";
+
+    @Inject
+    private EntityNameValidationManager entityNameValidationManager;
+
+    @Inject
+    @Named("local")
+    private EntityReferenceSerializer<String> entityReferenceSerializer;
+
+    @Override
+    public String getName()
+    {
+        return NAME;
+    }
+
+    @Override
+    public List<Event> getEvents()
+    {
+        LocalDocumentReference classReference = EntityNameValidationConfigurationSource.CLASS_REFERENCE;
+        LocalDocumentReference documentReference = EntityNameValidationConfigurationSource.DOC_REFERENCE;
+        RegexEventFilter configurationClassEventFilter = createRegexEventFilter(classReference);
+        RegexEventFilter configurationObjectEventFilter = createRegexEventFilter(documentReference);
+        return Arrays.asList(
+            new DocumentCreatedEvent(configurationClassEventFilter),
+            new DocumentUpdatedEvent(configurationClassEventFilter),
+            new DocumentDeletedEvent(configurationClassEventFilter),
+            new DocumentCreatedEvent(configurationObjectEventFilter),
+            new DocumentUpdatedEvent(configurationObjectEventFilter),
+            new DocumentDeletedEvent(configurationObjectEventFilter)
+        );
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        // Clear the cache of all the validation components for the wiki of the updated configuration.
+        String wikiName = ((XWikiDocument) source).getDocumentReference().getWikiReference().getName();
+        this.entityNameValidationManager.getAvailableEntityNameValidationsComponents()
+            .forEach(it -> it.cleanConfigurationCache(wikiName));
+    }
+
+    /**
+     * Creates a {@link RegexEventFilter} matching a local reference on any wiki. Schematically, for a local reference
+     * {@code XWiki.Doc}, this method created a {@link RegexEventFilter} with the following regex: {@code
+     * ".*:XWiki\.Doc"}.
+     *
+     * @param localDocumentReference the local document reference to filter on
+     * @return the even filter matching the local document reference events
+     */
+    private RegexEventFilter createRegexEventFilter(LocalDocumentReference localDocumentReference)
```

Comment:
Maybe this helper should be moved in a more reusable location since it looks like a very generic need.

Having a more specific method name like `anyLocalDocument` might help readability (as you probably noticed from my previous deleted comments :) )